### PR TITLE
Add PASTIS evaluation dataset

### PR DESCRIPTION
**Pull request by yawenzzzz**
- [x] Add PASTIS dataset
- [x] Process PASTIS-R (S2 + S1) dataset
- [x] Add PASTIS-R dataset
- [x] Make evaluation epoch configurable per task
- [x] Add test
- [x] Run experiment

NOTE: right now, I processed the PASTIS data similarly as in Presto, i.e., getting the monthly mean of Sentinel2, but this may not be great given the potential cloud coverage, but there's no metadata on cloud coverage, so I don't have a very good solution for it. For comparison purpose, I think using mean is fine.

### CODE REVIEW

File: helios/evals/datasets/constants.py [hunk 1]
```diff
@@ -18,24 +18,30 @@
...
 
+EVAL_S1_BAND_NAMES = [
+    "vv",
+    "vh",
+]
 
-EVAL_S1_BAND_NAMES = ["vv", "vh"]
+def _eval_s2_band_index_from_helios_name(helios_name: str) -> int:
+    for idx, band_name in enumerate(EVAL_S2_BAND_NAMES):
+        if helios_name.endswith(band_name.split(" ")[0][-2:]):
```
Comment (favyen2) [inline helios/evals/datasets/constants.py:29]: Suggest to add comment here to explain why this is necessary perhaps with an example of Helios band vs band name in EVAL_S2_BAND_NAMES (and why can't just change EVAL_S2_BAND_NAMES).

File: helios/evals/datasets/pastis_dataset.py [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+        """Process the PASTIS-R dataset."""
+        with open(self.data_dir / "metadata.geojson") as f:
+            meta_data = json.load(f)
+
+        all_data: dict[str, dict[str, list[torch.Tensor]]] = {
+            f"fold_{i}": {"s2_images": [], "s1_images": [], "months": [], "targets": []}
+            for i in range(1, 6)
+        }
+
+        doesnt_have_twelve = 0
```
Comment (favyen2) [inline helios/evals/datasets/pastis_dataset.py:220]: Would be helpful to have a comment here saying that some PASTIS examples don't have Sentinel-2 images from 12 distinct months.

File: helios/evals/datasets/pastis_dataset.py [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        """Aggregate images into monthly averages."""
+        months_dict = dict[str, list[torch.Tensor]]()
+        for m in self.all_months:
+            months_dict[m] = []
+
+        for idx, date in dates.items():
+            month = str(date)[:6]
+            img = torch.tensor(images[int(idx)], dtype=torch.float32)
+            if images.shape[1] == 10:
```
Comment (favyen2) [inline helios/evals/datasets/pastis_dataset.py]: If it is always expected to have 10 bands here, then suggest verifying that (throw ValueError) and always imputing instead of like this.

If some images have 10 bands but others have 13 bands, then a comment explaining it would be helpful.

File: helios/evals/datasets/pastis_dataset.py [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+            s2_image = self.normalizer_computed.normalize(
+                Modality.SENTINEL2_L2A, s2_image
+            )
+            if self.is_multimodal:
+                s1_image = self.normalizer_computed.normalize(
+                    Modality.SENTINEL1, s1_image
+                )
+
+        timestamps = []
+        # A little hack to get the correct year for the timestamps
```
Comment (favyen2) [inline helios/evals/datasets/pastis_dataset.py]: Why not store both year and month in the months.pt instead of just month? It seems there is some overlap in the months.

Otherwise suggest to all_months to 2018-09 to 2019-08 so there is no overlap and this code here will always work. But it seems that storing the year should be preferred so there is no need to guess when reading it.

File: helios/evals/datasets/pastis_dataset.py [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+            masked_sample = MaskedHeliosSample.from_heliossample(
+                HeliosSample(
+                    sentinel2_l2a=torch.tensor(s2_image).float(),
+                    timestamps=timestamps,
+                )
+            )
+        return masked_sample, labels.long()
+
+
+# if __name__ == "__main__":
```
Comment (favyen2) [inline helios/evals/datasets/pastis_dataset.py]: I think either uncomment or remove.

File: helios/train/callbacks/evaluator_callback.py [hunk 6]
```diff
@@ -77,10 +78,7 @@ def _get_embeddings(
 
     def val(self) -> float:
         """Validate the model on the downstream task."""
-        # try:
```
Comment (favyen2) [inline helios/train/callbacks/evaluator_callback.py:80]: Remove the commented code added here.

File: helios/train/callbacks/evaluator_callback.py [hunk 7]
```diff
@@ -128,33 +126,29 @@ def val(self) -> float:
             raise ValueError(f"Unrecognized task type: {self.config.task_type}")
         logger.info(f"Downstream evaluator {self.dataset} score: {val_result}")
         return val_result
-        # except Exception as e:
```
Comment (favyen2) [inline helios/train/callbacks/evaluator_callback.py:131]: Remove the commented code added here.

Comment (yawenzzzz) [reply helios/train/callbacks/evaluator_callback.py:131]: Fixed.

File: helios/train/callbacks/evaluator_callback.py (hunk 6 above)
Comment (yawenzzzz) [reply helios/train/callbacks/evaluator_callback.py:80]: Fixed.

File: helios/evals/datasets/pastis_dataset.py (hunk 5 above)
Comment (yawenzzzz) [reply helios/evals/datasets/pastis_dataset.py]: Fixed, I removed it.

File: helios/evals/datasets/constants.py (hunk 1 above)
Comment (yawenzzzz) [reply helios/evals/datasets/constants.py:29]: These names are adopted from Geobench datasets, their band stats are saved in json files, for example, /weka/dfive-default/presto-geobench/dataset/geobench/classification_v1.0/m-eurosat/band_stats.json, and the Geobench dataset class will load the json file get the normalization stats. We use the same S2 band names for all eval datasets that require S2 data for consistency. I added a comment on this.

File: helios/evals/datasets/pastis_dataset.py (hunk 2 above)
Comment (yawenzzzz) [reply helios/evals/datasets/pastis_dataset.py:220]: Sure, added a comment on this, and the results is 0 meaning that all samples have the 12 months' data.

File: helios/evals/datasets/pastis_dataset.py (hunk 3 above)
Comment (yawenzzzz) [reply helios/evals/datasets/pastis_dataset.py]: It always has 10 bands, I added a check there to verify if it's sentinel2 and if the band numbers is equal to 10.

File: helios/evals/datasets/pastis_dataset.py (hunk 4 above)
Comment (yawenzzzz) [reply helios/evals/datasets/pastis_dataset.py]: OH nice catch! Yea, this is unnecessary as I can easily get the year from the year-month str. I updated this part.
